src/shelly_hap_rgb.cpp

```diff
@@ -330,42 +374,59 @@ Status RGBWLight::SetState(const std::string &state_json) {
              "{state: %B, brightness: %d, hue: %d, saturation: %d}", &state,
              &brightness, &hue, &saturation);
 
-  if (cfg_->state != state) {
-    cfg_->state = state;
-    dirty_ = true;
-  }
-  if (cfg_->brightness != brightness) {
-    cfg_->brightness = brightness;
-    dirty_ = true;
-  }
-  if (cfg_->hue != hue) {
-    cfg_->hue = hue;
-    dirty_ = true;
-  }
-  if (cfg_->saturation != saturation) {
-    cfg_->saturation = saturation;
-    dirty_ = true;
-  }
-
-  if (dirty_) {
-    SetOutputState("RPC");
-  }
+  UpdateOnOff(state, "RPC");
+  SetHsv(hue, saturation, brightness, "RPC");
 
   return Status::OK();
 }
 
+void RGBWLight::ResetAutoOff() {
+  auto_off_timer_.Reset(cfg_->auto_off_delay * 1e3, 0);
```

Comment:
i can read scientific notation but please... 1000 is much more readable :)

Reply:
Ok I will change it to 1000, but would you say this also for 1000000? I saw it several times :-)

Reply:
i didn't give it much thought but in all cases new code should be kept consistent with existing code, unless old code is an obvious antipattern (in which case it should be refactored).
this one is not a gross antipattern, more a matter of taste. i don't see pressing need for changing it, i don't find 1e6 vastly more readable, maybe because while i'm familiar with it, i'm not using scientific notation every day and it takes a moment to realize it's a number, whereas "1000000" is burned into my brain already.